pkg/bpf/map_linux.go

```diff
@@ -318,6 +319,9 @@ func (m *Map) DumpWithCallback(cb DumpCallback) error {
 	}
 
 	if err := GetFirstKey(m.fd, unsafe.Pointer(&nextKey[0])); err != nil {
+		if err == io.EOF {
+			return nil
+		}
 		return nil
```

Comment:
I'm surprised the linter didn't complain because the if seems unnecessary here.

Reply:
Good catch

Reply:
that does seem a bit pointless...  What I mean is EOF is an expected error so return nil. Otherwise we want to return the error because its something else.

Reply:
Line 325 should be 'return err' maybe?